### PR TITLE
Stuff

### LINKED ISSUE #54: As a student, I want to edit a module's description
so that I can modify the module’s description

### LINKED ISSUE #58: As a student, I want to delete a module
so that I can use the App for different semesters

### PR DESCRIPTION
**Pull request by NNpanpan**
- Mod delete/edit can use module code: Mod edit must have the /code prefix, while Mod delete can just type the code.
- Mod delete will delete related tasks (and facilitators with single module) in O(n) time :>
- Cmd group command: kinda trivia ...
- Regex for ModuleCode: CS2030, CS2040S, PCS1101, PCS1101T are all valid - kinda like NUS official mod codes.

### CODE REVIEW

File: src/main/java/seedu/address/model/facilitator/UniqueFacilitatorList.java [hunk 1]
```diff
@@ -89,29 +89,45 @@ public void remove(Facilitator toRemove) {
...
+                })
+                .map(facilitator -> {
+                    Set<ModuleCode> moduleCodes = new HashSet<>(facilitator.getModuleCodes());
+                    moduleCodes.remove(toRemove);
+                    return new Facilitator(facilitator.getName(), facilitator.getPhone(),
+                            facilitator.getEmail(), facilitator.getOffice(), moduleCodes);
+                })
+                .forEach(replacementList::add);
+
+        setFacilitators(replacementList);
```
Comment (zixinn) [inline src/main/java/seedu/address/model/facilitator/UniqueFacilitatorList.java]: What about facilitators whose module code does not contain toRemove? I think have to add them to the replacementList too

File: src/main/java/seedu/address/logic/commands/calendar/CalCommand.java [hunk 2]
```diff
@@ -14,6 +14,8 @@
 public abstract class CalCommand extends Command {
 
     public static final List<String> ALL_COMMAND_WORDS = List.of(COMMAND_WORD_VIEW, COMMAND_WORD_FIND);
+    public static final String CAL_VIEW_FORMAT = COMMAND_GROUP_CAL + " " + COMMAND_WORD_VIEW + " this/week";
```
Comment (alushingg) [inline src/main/java/seedu/address/logic/commands/calendar/CalCommand.java]: So now view is going to show this and any week ah. Or is this typo.

File: src/main/java/seedu/address/logic/commands/module/ModuleDeleteCommand.java [hunk 3]
```diff
@@ -22,31 +23,56 @@
     public static final String MESSAGE_USAGE = COMMAND_GROUP_MOD + " " + COMMAND_WORD_DELETE
             + ": Deletes the module identified by the index number used in the displayed module list.\n"
             + "Parameters: INDEX (must be a positive integer)\n"
-            + "Example: " + COMMAND_GROUP_MOD + " " + COMMAND_WORD_DELETE + " 1";
+            + "Example: " + COMMAND_GROUP_MOD + " " + COMMAND_WORD_DELETE + " 1\n"
+            + "Parameters: MODULE_CODE (must be a valid module code)\n"
```
Comment (zixinn) [inline src/main/java/seedu/address/logic/commands/module/ModuleDeleteCommand.java:27]: I think can update ModuleCommand.java to reflect the change in format.

File: src/main/java/seedu/address/logic/commands/module/ModuleEditCommand.java [hunk 4]
```diff
@@ -30,16 +30,22 @@
             + "by the index number used in the displayed module list. "
             + "Existing values will be overwritten by the input values.\n"
             + "Parameters: INDEX (must be a positive integer) "
-            + "[" + PREFIX_MODULE_CODE + " MOD_CODE] "
-            + "[" + PREFIX_DESCRIPTION + " DESCRIPTION]\n"
+            + PREFIX_DESCRIPTION + " DESCRIPTION\n"
             + "Example: " + COMMAND_GROUP_MOD + " " + COMMAND_WORD_EDIT + " 1 "
-            + PREFIX_DESCRIPTION + "new description";
+            + PREFIX_DESCRIPTION + " new description\n"
+            + "Parameters: " + PREFIX_MODULE_CODE + " MOD_CODE "
```
Comment (zixinn) [inline src/main/java/seedu/address/logic/commands/module/ModuleEditCommand.java:36]: I think can update ModuleCommand.java to reflect the change in format.

File: src/main/java/seedu/address/logic/commands/module/ModuleDeleteCommand.java [hunk 5]
```diff
@@ -22,31 +23,56 @@
...
+            for (Module mod : lastShownList) {
+                if (mod.getModuleCode().equals(moduleCode)) {
+                    moduleToDelete = mod;
+                    break;
+                }
+            }
         }
 
-        Module moduleToDelete = lastShownList.get(targetIndex.getZeroBased());
+        assert moduleToDelete != null;
```
Comment (alushingg) [inline src/main/java/seedu/address/logic/commands/module/ModuleDeleteCommand.java]: Maybe can throw exception instead like this module does not exist

File: src/main/java/seedu/address/logic/commands/calendar/CalCommand.java (hunk 2 above)
Comment (NNpanpan) [reply src/main/java/seedu/address/logic/commands/calendar/CalCommand.java]: yeah it's a typo

File: src/main/java/seedu/address/model/facilitator/UniqueFacilitatorList.java (hunk 1 above)
Comment (NNpanpan) [reply src/main/java/seedu/address/model/facilitator/UniqueFacilitatorList.java]: whoops forgot that part

File: src/main/java/seedu/address/logic/commands/module/ModuleDeleteCommand.java (hunk 5 above)
Comment (NNpanpan) [reply src/main/java/seedu/address/logic/commands/module/ModuleDeleteCommand.java]: yup yup

File: src/main/java/seedu/address/logic/commands/module/ModuleEditCommand.java [hunk 6]
```diff
@@ -54,18 +60,39 @@ public ModuleEditCommand(Index index, ModuleEditCommand.EditModuleDescriptor edi
...
+            for (Module mod : lastShownList) {
+                if (mod.getModuleCode().equals(moduleCode)) {
+                    moduleToEdit = mod;
+                    break;
+                }
+            }
         }
 
-        Module moduleToEdit = lastShownList.get(index.getZeroBased());
+        assert moduleToEdit != null;
```
Comment (zixinn) [inline src/main/java/seedu/address/logic/commands/module/ModuleEditCommand.java]: I think you can throw exception instead when the module code does not exist.

Comment (NNpanpan) [reply src/main/java/seedu/address/logic/commands/module/ModuleEditCommand.java]: yeah i somehow thought the exception had been caught somewhere else
and then realized nope

**Pull request merged**